File: lib/services/mock/mock_client_service.dart
import 'package:dartz/dartz.dart';
import '../../core/errors/failures.dart';
import '../../core/network/dio_client.dart';
import '../../models/client_model.dart';
import '../client_service.dart';
import 'mock_data.dart';

class MockClientService extends ClientService {
  MockClientService(DioClient dioClient) : super(dioClient);

  @override
  Future<Either<Failure, List<ClientModel>>> searchClients(String query) async {
    await Future.delayed(const Duration(milliseconds: 400));

    final q = query.toLowerCase();
    final results = MockData.clients
        .where((c) =>
            c.code.toLowerCase().contains(q) ||
            c.nom.toLowerCase().contains(q) ||
            (c.ville?.toLowerCase().contains(q) ?? false))
        .toList();

    return Right(results);
  }

  @override
  Future<Either<Failure, ClientModel>> getClientByCode(String code) async {
    await Future.delayed(const Duration(milliseconds: 300));

    final client = MockData.clients.where((c) => c.code == code).firstOrNull;
    if (client == null) {
      return const Left(ServerFailure(message: 'Client introuvable', statusCode: 404));
    }
    return Right(client);
  }
}

File: olympia_admin/src/services/mock/mockClientService.js
import { MOCK_CLIENTS } from '../../data/mockData.js';

const mockClientService = {
  async search(query = '') {
    await _delay(200);
    const q = query.toLowerCase();
    const results = q
      ? MOCK_CLIENTS.filter((c) => c.nom.toLowerCase().includes(q) || c.code.toLowerCase().includes(q) || c.ville.toLowerCase().includes(q))
      : MOCK_CLIENTS;
    return { data: results, error: null };
  },

  async getByCode(code) {
    await _delay(150);
    const client = MOCK_CLIENTS.find((c) => c.code === code);
    if (!client) return { data: null, error: { message: 'Client introuvable', code: 404 } };
    return { data: client, error: null };
  },
};

function _delay(ms) { return new Promise((r) => setTimeout(r, ms)); }

export default mockClientService;
